src/components/input/input.jsx

```diff
@@ -1,37 +1,91 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import cautionSvg from '../../svgs/icon-caution-fill.svg';
+import Icon from '../icon/icon.jsx';
 import styles from './input.css';
-import labelPropType from './label-prop-type';
 
-/**
- * Base input component. Accepts any prop that an input can take.
- * @see [RFC](https://github.com/mavenlink/rfc/pull/75)
- * @see [Design spec](https://zpl.io/295Yp7p)
- */
-export default function Input({ 'aria-labelledby': ariaLabelledBy, className, id, ...rest }) {
+function getLabelClassName(className, invalid) {
+  if (className) return className;
+  return invalid ? styles['invalid-label'] : styles.label;
+}
+
+function getClassName(className, invalid) {
+  if (className) return className;
+  return invalid ? styles['invalid-input'] : styles.input;
+}
+
+export default function Input(props) {
   return (
-    <input
-      {...rest}
-      aria-labelledby={ariaLabelledBy}
-      className={className}
-      id={id}
-    />
+    <React.Fragment>
+      <label htmlFor={props.id} className={getLabelClassName(props.cssLabel, props.invalid)}>
+        {props.label}
+      </label>
+      <div className={props.cssContainer}>
+        <input
+          className={getClassName(props.className, props.invalid)}
+          disabled={props.disabled}
+          id={props.id}
+          maxLength={props.maxLength}
+          name={props.name}
+          onBlur={props.onBlur}
+          onChange={props.onChange}
+          onFocus={props.onFocus}
+          onInput={props.onInput}
+          onKeyDown={props.onKeyDown}
+          placeholder={props.placeholder}
+          readOnly={props.readOnly}
+          required={props.required}
+          type={props.type}
+          value={props.value}
+        />
+        {props.invalid && <Icon className={styles['invalid-icon']} currentColor="caution" name={cautionSvg.id} />}
+      </div>
+    </React.Fragment>
   );
 }
 
 Input.propTypes = {
-  /**
-   * Reference to the ID of a labelling element. Useful for tables where the column header can serve as
-   * the label for the input. Either this or 'id' must be present.
-   */
-  'aria-labelledby': labelPropType,
   className: PropTypes.string,
-  /** ID used by a corresponding label element. Either this or 'aria-labelledby' must be present. */
-  id: labelPropType,
+  cssContainer: PropTypes.string,
+  cssLabel: PropTypes.string,
+  disabled: PropTypes.bool,
+  id: PropTypes.string.isRequired,
+  invalid: PropTypes.bool,
+  label: PropTypes.string.isRequired,
+  maxLength: PropTypes.number,
+  name: PropTypes.string,
+  onBlur: PropTypes.func,
+  onChange: PropTypes.func,
+  onFocus: PropTypes.func,
+  onInput: PropTypes.func,
+  onKeyDown: PropTypes.func,
+  placeholder: PropTypes.string,
+  readOnly: PropTypes.bool,
+  required: PropTypes.bool,
+  type: PropTypes.oneOf([
```

Comment:
We should consider making these `type`s separate components.

Reply:
That's an interesting point! 

I think that is 100% the right thing when dealing with Input types that are drastically different than a text field.

I am not convinced we need to do it for all types though. I think we might want to get more data on this but for now this is a reasonably small list!

Reply:
Fair enough! We can always iterate if we see this bloating the component down the line, I'm mainly worried about "god" components, and I think this is still a long way from that.